### DISCUSSION TITLE
How do i append URL GET parameters?

### DISCUSSION THREAD

**Discussion by livevsonline** [TanStack/router#507] [Q&A] [answered]
Hi

I have a form, and i would like to append dynamic parameters to the query string when the fields on the form change

How would i go about that with Tanstack-Router?

The documentation is still missing parts, it is also still beta soo :)

I found defaultStrinfySearch which perfectly creates an URL search query string.
I found createBrowserHistory, which perfectly creates browser history with my query params.

However i cant find anything in the autocompleete imports from @tanstack/react-router to push it to the query string?

I want to push URL parameters without refresing the page, just pushing them  to the url for bookmarks etc

Anyone able to help me out ? :)

**Comment by tannerlinsley** [comment, marked as answer]
Yep. You'll want to check out this example: https://tanstack.com/router/v1/docs/examples/react/kitchen-sink-single-file?file=src%2Fmain.tsx and look for a calls to `navigate()`. You can use the `navigate` function to update the URL without changing paths by simply not passing a new path and only passing the `search` option. If you don't want to make a new entry in the history stack, you can also pass `replace: true`.

**Reply by tannerlinsley** [reply]
```typescript
navigate({
  search: (old) => ({
    ...old,
    notes: notes ? notes : undefined,
  }),
  replace: true,
})
```

**Reply by livevsonline** [reply]
> Yep. You'll want to check out this example: https://tanstack.com/router/v1/docs/examples/react/kitchen-sink-single-file?file=src%2Fmain.tsx and look for a calls to `navigate()`. You can use the `navigate` function to update the URL without changing paths by simply not passing a new path and only passing the `search` option. If you don't want to make a new entry in the history stack, you can also pass `replace: true`.

The sample comes with this:

Unable to run Embedded Project

Looks like this project is being embedded without proper isolation headers.

Also typescript is not at all happy with that code, it complains that the parameters is not compatible with:
is not assignable to parameter of type 'ParamsId | ParamsCombo

Could you explain what the different variables is and what it holds ? :)

**Reply by livevsonline** [reply]
> ```ts
> navigate({
>   search: (old) => ({
>     ...old,
>     notes: notes ? notes : undefined,
>   }),
>   replace: true,
> })
> ```

I'm getting quite a lot of typescript errors from this tho? :)
